src/avatar_action.cpp

```diff
@@ -498,6 +499,13 @@ void avatar_action::swim( map &m, avatar &you, const tripoint &p )
         add_msg( m_warning, _( "You cannot board a vehicle while mounted." ) );
         return;
     }
+    if( m.veh_at( p ).part_with_feature( VPFLAG_BOARDABLE, true ) ) {
```

Comment:
```suggestion
    if( const auto vp = m.veh_at( p ).part_with_feature( VPFLAG_BOARDABLE, true ) ) {
         if( vp->vehicle().handle_potential_theft( dynamic_cast<player *>( &you ) ) ) {
```
Don't query the map again for information you already have.